packages/css/src/transformCss.ts

```diff
@@ -263,18 +278,43 @@ class Stylesheet {
       });
     }
 
-    return this.localClassNameRegex
-      ? transformedSelector.replace(
-          this.localClassNameRegex,
-          (_, className, index) => {
-            if (index > 0 && transformedSelector[index - 1] === '.') {
-              return className;
-            }
+    if (this.localClassNamesMap.has(transformedSelector)) {
+      return `.${cssesc(transformedSelector, { isIdentifier: true })}`;
+    }
+
+    const results = this.localClassNamesSearch.search(transformedSelector);
 
-            return `.${cssesc(className, { isIdentifier: true })}`;
-          },
-        )
-      : transformedSelector;
+    let lastReplace = transformedSelector.length;
+
+    // Perform replacements backwards to simplify index handling
+    for (let i = results.length - 1; i >= 0; i--) {
+      const [endIndex, [firstMatch]] = results[i];
+      const startIndex = endIndex - firstMatch.length + 1;
+
+      if (lastReplace <= startIndex) {
+        // Class names can be substrings of other class names
+        // e.g. '_1g1ptzo1' and '_1g1ptzo10'
+        // If the last replaced index is <= startIndex, then
+        // this is the case and this replace should be skipped
+        continue;
+      }
+
+      lastReplace = startIndex;
+
+      // If class names already starts with a '.' then skip
+      if (transformedSelector[startIndex - 1] !== '.') {
+        transformedSelector = replaceBetweenIndexes(
+          transformedSelector,
+          startIndex,
+          endIndex + 1,
+          `.${cssesc(firstMatch, {
+            isIdentifier: true,
+          })}`,
```

Comment:
```suggestion
          `.${cssesc(firstMatch, { isIdentifier: true, })}`,
```

Also, since this logic is now in [two places](https://github.com/seek-oss/vanilla-extract/blob/transform-perf/packages/css/src/transformCss.ts#L282) do you think it's worth creating an `escapeIdentifier` function to make sure they don't drift apart?